code/modules/projectiles/guns/specialist.dm

```diff
@@ -489,71 +489,45 @@
 //-------------------------------------------------------
 //GRENADE LAUNCHER
 
-/obj/item/weapon/gun/launcher/m92
-	name = "\improper T-26 grenade launcher"
-	desc = "A heavy, 6-shot grenade launcher used by the TerraGov Marine Corps for area denial and big explosions."
-	icon_state = "m92"
-	item_state = "m92"
-	max_shells = 6 //codex
-	caliber = CALIBER_40MM //codex
-	load_method = SINGLE_CASING //codex
+/obj/item/weapon/gun/launcher
 	w_class = WEIGHT_CLASS_BULKY
+	gun_skill_category = GUN_SKILL_FIREARMS
+	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_WIELDED_FIRING_ONLY|GUN_AMMO_COUNTER
 	throw_speed = 2
 	throw_range = 10
-	force = 5.0
-	wield_delay = 0.6 SECONDS
+	force = 5
+	caliber = CALIBER_40MM //codex
+	load_method = SINGLE_CASING //codex
 	fire_sound = 'sound/weapons/guns/fire/grenadelauncher.ogg'
 	fire_rattle = 'sound/weapons/guns/fire/grenadelauncher.ogg'
 	cocked_sound = 'sound/weapons/guns/interact/m92_cocked.ogg'
-	var/list/grenades = list()
-	var/max_grenades = 6
-	aim_slowdown = 1
 	general_codex_key = "explosive weapons"
-	attachable_allowed = list(
-		/obj/item/attachable/magnetic_harness,
-		/obj/item/attachable/scope/mini,
-	)
-
-	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_WIELDED_FIRING_ONLY|GUN_AMMO_COUNTER
-	gun_skill_category = GUN_SKILL_FIREARMS
-	var/datum/effect_system/smoke_spread/smoke
-	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 18,"rail_x" = 14, "rail_y" = 22, "under_x" = 19, "under_y" = 14, "stock_x" = 19, "stock_y" = 14)
-
-	fire_delay = 1.8 SECONDS
-
-
-/obj/item/weapon/gun/launcher/m92/Initialize()
-	. = ..()
-	for(var/i in 1 to max_grenades)
-		grenades += new /obj/item/explosive/grenade(src)
-
-/obj/item/weapon/gun/launcher/m92/update_icon(mob/user)
-	update_item_state(user)
-	update_mag_overlay(user)
-
-/obj/item/weapon/gun/launcher/m92/examine_ammo_count(mob/user)
-	if(!length(grenades) || (get_dist(user, src) > 2 && user != loc))
-		return
-	to_chat(user, span_notice(" It is loaded with <b>[length(grenades)] / [max_grenades]</b> grenades."))
-
-
-/obj/item/weapon/gun/launcher/m92/attackby(obj/item/I, mob/user, params)
-	if(!istype(I, /obj/item/explosive/grenade))
-		return ..()
-
-	if(length(grenades) >= max_grenades)
-		to_chat(user, span_warning("[src] cannot hold more grenades!"))
-		return
-
-	if(!user.transferItemToLoc(I, src))
-		return
+	var/list/grenades = list()
+	var/max_grenades
 
-	grenades += I
-	playsound(user, 'sound/weapons/guns/interact/shotgun_shell_insert.ogg', 25, 1)
-	to_chat(user, span_notice("You put [I] in [src]."))
-	to_chat(user, span_info("Now storing: [grenades.len] / [max_grenades] grenades."))
+/obj/item/weapon/gun/launcher/proc/fire_grenade(atom/target, mob/user)
+	last_fired = world.time
+	var/obj/item/explosive/grenade/G = grenades[1]
```

Comment:
Change the name here. Single letter vars make code confusing to read. Name it grenade or something.